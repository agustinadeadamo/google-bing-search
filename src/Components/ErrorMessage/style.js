/**
 * @desc Dependencies
 */
import styled from "styled-components";

export const ErrorMessageContainer = styled.div`
    width: 100%;
    text-align: center;

    .button-error-message-container {
        width: 100%;
        max-width: 300px;
        margin: 50px auto;
    }
`;