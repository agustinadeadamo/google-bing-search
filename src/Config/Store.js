/**
 * @desc Dependencies
 */
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

/**
 * @desc Middlewares
 */
import Middlewares from './Middlewares';

/**
 * @desc Reducers
 */
import Reducers from 'Reducer/Reducer';

/**
 * @desc Store
 */
const Store = createStore(Reducers, composeWithDevTools(applyMiddleware(...Middlewares)));

export default Store;
