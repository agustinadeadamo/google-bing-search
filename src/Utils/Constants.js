/* Search mode */
export const SearchMode = {
    "GOOGLE" : 'Google',
    "BING" : 'Bing',
    "GOOGLE_BING" : 'Goolge & Bing',
};

export const MessageError = {
    'SERVER_ERROR': 'There was an error in the server',
    'APP_ERROR': 'There was an error in the app',
};

export const Apikeys = {
    'BING_KEY' : '1ef3364ab9ce48b1b180b109e6aab6fb',
    'GOOGLE_KEY': 'AIzaSyB5Mqgm1gE_sAvmQToGkD2fGnOw-Pl1m5w',
    'GOOGLE_CX': '18ca2e163998cf02d'
}

export const Endpoints = {
    'GOOGLE_SEARCH' : 'https://www.googleapis.com/customsearch/v1',
    'BING_SEARCH': 'https://api.bing.microsoft.com/v7.0/search'
}

