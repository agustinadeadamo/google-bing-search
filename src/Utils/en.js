export const English = {
    'ADD_VALUE_TO_SEARCH' : 'Add a value to search',
    'SELECT_OPTION' : 'Select an option',
    'SUBMIT' : 'Submit',
    'GOOGLE_BING_SEARCH' : 'Google & Bing Search',
}