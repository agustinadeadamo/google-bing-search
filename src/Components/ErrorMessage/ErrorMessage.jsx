/**
 * @desc Dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';

/**
 * @desc Styles
 */
import { ErrorMessageContainer } from './style';

/**
 * @desc Actions
 */
import { hideError } from 'Actions/ErrorActions';

/**
 * @desc Styled Components
 */
import { ButtonPrimary } from 'Mainstyles/Components.js';

const ErrorMessage = (props) => {

    // Destructuring props
    const {
        message,
    } = props;

    // Store
    const dispatch = useDispatch()

    /**
     * Function that is executed when on click in button hides error 
     */
    const onClickAcceptButton = () => {

        try {
            dispatch(hideError(false))
        } catch(error) {
            console.log(error)
        }

    }

    return (
        <ErrorMessageContainer data-testid="error-message-component">
                <p data-testid="message-error">{ message }</p>
                <div className="button-error-message-container">
                    <ButtonPrimary data-testid="button-error-message" onClick={() => onClickAcceptButton()}>Accept</ButtonPrimary>''
                </div>
        </ErrorMessageContainer>
    );
};

ErrorMessage.propTypes = {
    message: PropTypes.string.isRequired,
};

ErrorMessage.defaultProps = {
    message: ''
};

export default ErrorMessage;
