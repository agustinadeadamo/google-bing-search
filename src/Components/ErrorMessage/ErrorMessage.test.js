/**
 * @desc Dependencies
 */
import { create } from 'react-test-renderer';
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

/**
 * @desc Config
 */
import Store from 'Config/Store';

/**
 * @desc Utils
 */
import { MessageError } from 'Utils/Constants';

/**
 * @desc Components
 */
import ErrorMessage from './ErrorMessage';

describe('[Error Message Component]',() => {

    it('Renders correctly', () => {

        const {queryByTestId} = render(
            <Provider store={Store}>
                <ErrorMessage message={MessageError.APP_ERROR}/>
            </Provider>
        );

        expect(queryByTestId('error-message-component')).toBeTruthy();
    
    });

    it('Matches snapshot', () => {
        
        const input = create(
            <Provider store={Store}>
                <ErrorMessage message={MessageError.APP_ERROR}/>
            </Provider>
        ).toJSON;

        expect(input).toMatchSnapshot();
        
    });

    it('Shows app error message', () => {

        const {getByText} = render(
            <Provider store={Store}>
                <ErrorMessage message={MessageError.APP_ERROR}/>
            </Provider>
        );

        expect(getByText(MessageError.APP_ERROR)).toBeInTheDocument();
    
    });

    it('Shows server error message', () => {

        const {getByText} = render(
            <Provider store={Store}>
                <ErrorMessage message={MessageError.SERVER_ERROR}/>
            </Provider>
        );

        expect(getByText(MessageError.SERVER_ERROR)).toBeInTheDocument();
    
    });
});