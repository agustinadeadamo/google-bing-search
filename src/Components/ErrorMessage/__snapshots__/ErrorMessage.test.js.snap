// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Error Message Component] Matches snapshot 1`] = `[Function]`;
